### PR TITLE
feat(server): added status api

### LINKED ISSUE #5: feat: add status rest api

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✔                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

Closes #5 

Further  information:

### CODE REVIEW

File: openapi3.yaml [hunk 1]
```diff
@@ -9,31 +9,37 @@ info:
...
         200:
           description: OK
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/resource'
+                $ref: '#/components/schemas/status'
     post:
-      operationId: createResource
+      operationId: updateStatus
```
Comment (asafMasa) [inline openapi3.yaml:26]: update is usually a **put** message

File: tests/integration/status/status.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      expect(saveMock).toHaveBeenCalledTimes(1);
+      expect(saveMock).toHaveBeenCalledWith(watchingSetting);
+      expect(response.status).toBe(httpStatusCodes.OK);
+
+      const status = response.body as IStatus;
+      expect(status.isWatching).toEqual(true);
+    });
+  });
+  describe('Bad Path', function () {
+    // All requests with status code of 400
```
Comment (asafMasa) [inline tests/integration/status/status.spec.ts:53]: ?

File: tests/integration/status/status.spec.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  });
+  describe('Sad Path', function () {
+    // All requests with status code 4XX-5XX
```
Comment (asafMasa) [inline tests/integration/status/status.spec.ts:56]: ?

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good Job

File: tests/integration/status/status.spec.ts (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply tests/integration/status/status.spec.ts:53]: forgot that. ill add it

File: tests/integration/status/status.spec.ts (hunk 3 above)
Comment (cl-shaharshavit-rnd) [reply tests/integration/status/status.spec.ts:56]: forgot that. ill add it

File: openapi3.yaml (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply openapi3.yaml:26]: this is designed to be upsert (add or update) should i change it?

**Review by asafMasa — APPROVED** [review]
🥇

**Pull request merged**